### DISCUSSION TITLE
math: LaTeX spacing commands parsed incorrectly

### DISCUSSION THREAD

**Discussion by briochemc** [community/community#17143] [Other Feature Feedback, Questions, & Ideas] [answered]
The most basic [spacing commands](http://docs.mathjax.org/en/latest/input/tex/extensions/textmacros.html?highlight=space#spacing-commands) don't work, e.g.,

```latex
$$a \, b$$

$$a \: b$$

$$a \> b$$

$$a \; b$$

$$a \! b$$

```

produces

$$a \, b$$

$$a \: b$$

$$a \> b$$

$$a \; b$$

$$a \! b$$

where it appears that all the backslashes `\` have been sanitized away, leaving unwanted symbols between $a$ and $b$. The expected result is just different sized white spaces between $a$ and $b$, as can be seen in the workaround suggested by @UlyssesZh below, that is, use double backslashes to escape the backslashes themselves:

```latex
$$a \\, b$$
```

produces the correct

$$a \\, b$$

This is not valid LaTeX, however, so it should be fixed if possible :) 

(Edit: probably related to https://github.com/github/feedback/discussions/16993.)

**Comment by Simran-B** [comment]
With backslash-escaped spaces:

```
$$a\ \,\ b$$

$$a\ \:\ b$$

$$a\ \>\ b$$

$$a\ \;\ b$$

$$a\ \!\ b$$
```

$$a\ \,\ b$$

$$a\ \:\ b$$

$$a\ \>\ b$$

$$a\ \;\ b$$

$$a\ \!\ b$$

So that's at least a workaround...

**Comment by nschloe** [comment]
The underlying problem here is that the Markdown parser (which runs first) "sanitizes" the math and removes all backslashes that go before non-letter characters. Not only spacing commands parse incorrectly, but also things like `\{` `\$` etc.

See https://nschloe.github.io/2022/05/20/math-on-github.html for a detailed write-up.

The fenced-math blocks syntax helps,
````markdown
```math
...
```
````
but doesn't work on existing repos like this one:
```math
a \, b

a \: b

a \> b

a \; b

a \! b
```
(Also doesn't help for inline math either, of course.)

**Reply by briochemc** [reply]
Yes, I saw the write up before, and actually tried to post this along your post about kerning (https://github.com/github/feedback/discussions/16997) but I could not for some reason so I opened this new one.

**Comment by UlyssesZh** [comment]
Markdown will first regard the backslash as escaping the symbol behind before MathJax comes in, so you need to use double backslash to make the backslash remaining to MathJax:
```md
$$a \\, b$$
```
This produces
$$a \\, b$$

**Reply by Simran-B** [reply]
Isn't the comma supposed to be visible? Your example produces `a b`

**Reply by UlyssesZh** [reply]
@Simran-B The comma is not supposed to be visible. `\,` is a LaTeX command to produce a thin space (equivalent to `\thinspace`) as wide as .16667em.

**Reply by briochemc** [reply]
Yes, @UlyssesZh is correct. @Simran-B: the workaround is the same one you posted in https://github.com/github/feedback/discussions/16993 (escaping the backslash to protect them). Should workarounds like this be "marked as answer"?

**Reply by briochemc** [reply]
I will edit my post to clarify what's expected.

**Reply by UlyssesZh** [reply]
> Should workarounds like this be "marked as answer"?

I think no. It is merely a workaround. We can reach the true solution only when GitHub further updates the feature to fix it.

**Comment by nschloe** [comment, marked as answer]
A reasonable fix here is to use the new backtick-delimited syntax:

````markdown
$`a \, b`$

```math
a \, b
```
````

> $`a \, b`$
>
> ```math
> a \, b
> ```

**Reply by UlyssesZh** [reply]
Should this be marked as answer now?

**Reply by nschloe** [reply]
I would guess that simple $-$$ math will never be fixed in this regard, so probably yes.
